cmd/kubeadm/app/preflight/checks_test.go

```diff
@@ -185,129 +186,222 @@ func (pfct preflightCheckTest) Check() (warning, errorList []error) {
+			expected: true,
 		},
 		{
-			name: "Test Cert file exists if specified",
-			cfg: &kubeadmapi.InitConfiguration{
-				ClusterConfiguration: kubeadmapi.ClusterConfiguration{
-					Etcd: kubeadmapi.Etcd{External: &kubeadmapi.ExternalEtcd{CertFile: "/foo"}},
-				},
+			name: "File is exist, content is non-exist and not nil",
+			check: FileContentCheck{
+				Path:    f.Name(),
+				Content: []byte("Test FileContentCheck but this content is not exist"),
 			},
 			expected: false,
 		},
 		{
-			name: "Test Key file exists if specified",
-			cfg: &kubeadmapi.InitConfiguration{
-				ClusterConfiguration: kubeadmapi.ClusterConfiguration{
-					Etcd: kubeadmapi.Etcd{External: &kubeadmapi.ExternalEtcd{CertFile: "/foo"}},
-				},
+			name: "File is non-exist, content is not nil",
+			check: FileContentCheck{
+				Path:    "/dose/not/exist",
+				Content: []byte("Test FileContentCheck but this file is not exist"),
 			},
 			expected: false,
 		},
 		{
-			name: "Test APIEndpoint exists if exists",
-			cfg: &kubeadmapi.InitConfiguration{
-				LocalAPIEndpoint: kubeadmapi.APIEndpoint{AdvertiseAddress: "2001:1234::1:15"},
+			name: "File is non-exist, content is nil",
+			check: FileContentCheck{
+				Path:    "/dose/not/exist",
+				Content: nil,
 			},
 			expected: false,
 		},
 	}
+	if _, err = f.WriteString("Test FileContentCheck"); err != nil {
+		t.Fatalf("Failed to write to file: %v", err)
+	}
 	for _, rt := range tests {
-		// TODO: Make RunInitNodeChecks accept a ClusterConfiguration object instead of InitConfiguration
-		actual := RunInitNodeChecks(exec.New(), rt.cfg, sets.NewString(), rt.isSecondaryControlPlane, rt.downloadCerts)
-		if (actual == nil) != rt.expected {
+		_, output := rt.check.Check()
+		if (len(output) == 0) != rt.expected {
 			t.Errorf(
-				"failed RunInitNodeChecks:%v\n\texpected: %t\n\t  actual: %t\n\t error: %v",
+				"Failed FileContentCheck:%v\n\texpected: %t\n\t  actual: %t",
 				rt.name,
 				rt.expected,
-				(actual == nil),
-				actual,
+				(len(output) == 0),
 			)
 		}
 	}
 }
 
-func TestRunJoinNodeChecks(t *testing.T) {
+func TestDirAvailableCheck(t *testing.T) {
+	fileDir, err := ioutil.TempDir("", "dir-")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(fileDir)
 	var tests = []struct {
 		name     string
-		cfg      *kubeadmapi.JoinConfiguration
+		check    DirAvailableCheck
 		expected bool
 	}{
 		{
-			name:     "Check empty JoinConfiguration",
-			cfg:      &kubeadmapi.JoinConfiguration{},
-			expected: false,
+			name: "Directory exist and has nothing",
+			check: DirAvailableCheck{
+				Path: fileDir,
+			},
+			expected: true,
 		},
 		{
-			name: "Check TLS Bootstrap APIServerEndpoint IPv4 addr",
-			cfg: &kubeadmapi.JoinConfiguration{
-				Discovery: kubeadmapi.Discovery{
-					BootstrapToken: &kubeadmapi.BootstrapTokenDiscovery{
-						APIServerEndpoint: "192.168.1.15",
-					},
-				},
+			name: "Directory exist and has something",
+			check: DirAvailableCheck{
+				Path: "/tmp",
```

Comment:
there is no guarantee that /tmp has something during test execution.
you should create another temp dir that has a empty file.

Reply:
At the beginning of the function, I have created a directory under /tmp